Tests failing due to issues building wheels for `so3` and `pyssht`

Several jobs in the [last scheduled run of the tests workflow failed](https://github.com/astro-informatics/s2fft/actions/runs/14150808608). It looks like this is due to errors when attempting to build wheels for the `so3` and `pyssht` packages required for running the tests, with for some of the jobs pre-built or cached wheels for these packages being picked up which is allowing the tests to run successfully. For both `pyssht` and `so3` the error seems to be caused by a CMake error when attempting to build `fftw`

```
       fftw/3.3.10: Aggregating env generators
      fftw/3.3.10: Calling build()
      fftw/3.3.10: apply_conandata_patches(): No patches defined in conandata
      fftw/3.3.10: CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/runner/.conan/data/fftw/3.3.10/_/_/build/76fb618e9f590ad1830f3b3f412b2dc9af0b6433/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/runner/.conan/data/fftw/3.3.10/_/_/package/76fb618e9f590ad1830f3b3f412b2dc9af0b6433" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" -DENABLE_FLOAT="ON" -DENABLE_LONG_DOUBLE="OFF" -DENABLE_QUAD_PRECISION="OFF" "/home/runner/.conan/data/fftw/3.3.10/_/_/build/76fb618e9f590ad1830f3b3f412b2dc9af0b6433/src"
      CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 has been removed from CMake.
      
        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.
      
        Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

which appears to be related to FFTW/fftw3#381

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -33,6 +33,8 @@ jobs:
           - os: macos-latest
             python-version: "3.8"
       fail-fast: false
+    env:
+      CMAKE_POLICY_VERSION_MINIMUM: 3.5
 
     steps:
     - name: Checkout Source

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -37,6 +37,9 @@ jobs:
     steps:
     - name: Checkout Source
       uses: actions/checkout@v4.2.2
+      with:
+        fetch-depth: 0
+        fetch-tags: true
 
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v5

